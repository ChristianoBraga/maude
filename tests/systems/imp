#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < imp.maude -no-banner -no-advise \
  > imp.out 2>&1

diff imp.expected imp.out > /dev/null 2>&1
