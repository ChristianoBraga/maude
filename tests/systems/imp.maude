set show timing off .

load ../../contrib/systems/imp.maude

rewrite < x = 3 ; ~> done | x |-> 0 > .

rewrite < sum(n, s) ~> done | n |-> 10 * s |-> 3 > .

rewrite < swap(x, y, t)       ~> done | x |-> 3 * y |-> 5 * t |-> 0 > .
rewrite < swap-in-place(x, y) ~> done | x |-> 3 * y |-> 5 > .
