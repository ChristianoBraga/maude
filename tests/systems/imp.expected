==========================================
rewrite in IMP-PROGRAMS : < x = 3 ; ~> done | x |-> 0 > .
rewrites: 3
result RedContext: < done | x |-> 1 :+ 1 :+ 1 >
==========================================
rewrite in IMP-PROGRAMS : < sum(n, s) ~> done | n |-> 10 * s |-> 3 > .
rewrites: 408
result RedContext: < done | n |-> 0 * s |-> 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+
    1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+
    1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+
    1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+ 1 :+
    1 :+ 1 :+ 1 >
==========================================
rewrite in IMP-PROGRAMS : < swap(x, y, t) ~> done | x |-> 3 * t |-> 0 * y |-> 5
    > .
rewrites: 20
result RedContext: < done | t |-> 1 :+ 1 :+ 1 * x |-> 1 :+ 1 :+ 1 :+ 1 :+ 1 * y
    |-> 1 :+ 1 :+ 1 >
==========================================
rewrite in IMP-PROGRAMS : < swap-in-place(x, y) ~> done | x |-> 3 * y |-> 5 > .
rewrites: 49
result RedContext: < done | x |-> 1 :+ 1 :+ 1 :+ 1 :+ 1 * y |-> 1 :+ 1 :+ 1 >
Bye.
